src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+    """
+    Dry spells
+
+    TODO: fill and link to ARC2
+    params using Sphinx so no
+    need for duplication
+
+    monitoring_start: Minimum date to load data from, either string
+        in ISO 8601 format, e.g. '2021-03-20' or `datetime.date` object.
+        Passed to `ARC2.date_min` as `monitoring_start - rolling_window`
+        days to ensure sufficient data for calculating rolling sum
+        available.
+    monitoring_end: Maximum date to load data from, either string
+        in ISO 8601 format, e.g. '2021-04-20' or `datetime.date` object.
+    :param agg_method: One of 'centroid',  'touching', or 'approximate_mask'.
+        If 'approximate_mask', the data is upsampled 4x before aggregating
+        using the 'centroid' method.
+    :param rolling_window: Number of days for rolling sum of precipitation.
+    :param rainfall_mm: Maximum precipitation during window to
+        classify as dry spell.
+    :param date_min: Minimum date to load data from, either string
+        in ISO 8601 format, e.g. '2021-03-20' or `datetime.date` object.
+    """
+
+    def __init__(
+        self,
+        country_iso3: str,
+        monitoring_start: Union[str, date],
+        monitoring_end: Union[str, date],
+        range_x: Tuple[str, str],
+        range_y: Tuple[str, str],
+        agg_method: str = "centroid",
+        rolling_window: int = 14,
+        rainfall_mm: int = 2,
+    ):
+        if not isinstance(monitoring_start, date):
+            monitoring_start = date.fromisoformat(monitoring_start)
+
+        if not isinstance(monitoring_end, date):
+            monitoring_end = date.fromisoformat(monitoring_end)
+
+        self.monitoring_start = monitoring_start
+        self.monitoring_end = monitoring_end
+        self.agg_method = agg_method
+        self.rolling_window = rolling_window
+        self.rainfall_mm = rainfall_mm
+
+        super().__init__(
+            country_iso3=country_iso3,
+            date_min=monitoring_start
+            - pd.to_timedelta(self.rolling_window - 1, unit="d"),
+            date_max=monitoring_end,
+            range_x=range_x,
+            range_y=range_y,
+        )
+
+    def aggregate_data(
+        self,
+        polygon_path: Union[Path, str] = None,
+        bound_col: str = None,
+        reprocess: bool = False,
+        redownload: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Get mean aggregation by admin boundary for the downloaded ARC2 data.
+        Outputs a csv with daily aggregated statistics. If data already
+        aggregated between `self.date_min` and `self.date_max`, returns
+        pre-aggregated data, otherwise aggregates additional data and joins
+        to aggregated main file.
+
+        :param polygon_path: Path to polygon file for clipping and aggregating
+            raster data.
+        :param bound_col: Column in polygon file to aggregate raster to.
+        :param reprocess: Boolean, if `True` reprocesses all raster data.
```

Comment:
The idea would be that reprocessing the data could be necessary if somehow the raw data changes, but maybe not worth adding in here because the likelihood of that happening is so low. Accounting for something that is a rare event I guess.